qiskit/circuit/quantumcircuit.py

```diff
@@ -3927,6 +3928,198 @@ def pauli(
+                in ``indexset`` but the values will not be assigned to ``body``.
+            indexset: A collection of integers to loop over.
+            body: The loop body to be repeatedly executed.
+            qubits: The circuit qubits over which the loop body should be run.
+            clbits: The circuit clbits over which the loop body should be run.
+            label: The string label of the instruction in the circuit.
+
+        Returns:
+            A handle to the instruction created.
+        """
+        # pylint: disable=cyclic-import
+        from qiskit.circuit.controlflow.for_loop import ForLoopOp
+
+        return self.append(ForLoopOp(loop_parameter, indexset, body, label), qubits, clbits)
+
+    def if_test(
+        self,
+        condition: Union[
+            Tuple[ClassicalRegister, int],
+            Tuple[Clbit, int],
+            Tuple[Clbit, bool],
+        ],
+        true_body: "QuantumCircuit",
+        qubits: Sequence[QubitSpecifier],
+        clbits: Sequence[ClbitSpecifier],
+        label: Optional[str] = None,
+    ) -> InstructionSet:
+        """Apply :class:`~qiskit.circuit.controlflow.IfElseOp` without a ``false_body``.
+
+        Args:
+            condition: A condition to be evaluated at circuit runtime which,
+                if true, will trigger the evaluation of ``true_body``. Can be
+                specified as either a tuple of a ``ClassicalRegister`` to be
+                tested for equality with a given ``int``, or as a tuple of a
+                ``Clbit`` to be compared to either a ``bool`` or an ``int``.
+            true_body: The circuit body to be run if ``condition`` is true.
+            qubits: The circuit qubits over which the if/else should be run.
+            clbits: The circuit clbits over which the if/else should be run.
+            label: The string label of the instruction in the circuit.
+
+        Raises:
+            CircuitError: If the provided condition references Clbits outside the
+                enclosing circuit.
+
+        Returns:
+            A handle to the instruction created.
+        """
+        # pylint: disable=cyclic-import
+        from qiskit.circuit.controlflow.if_else import IfElseOp
+
+        if isinstance(condition[0], ClassicalRegister):
+            if condition[0] not in self.cregs:
+                raise CircuitError(
+                    "The condition provided to QuantumCircuit.if_ was not "
+                    "found on the enclosing circuit. "
+                    f"condition: {condition}, QuantumCircuit.cregs: {self.cregs}."
+                )
+        elif isinstance(condition[0], Clbit):
+            if condition[0] not in self._clbit_indices:
+                raise CircuitError(
+                    "The condition provided to QuantumCircuit.if_ was not "
+                    "found on the enclosing circuit. "
+                    f"condition: {condition}, QuantumCircuit.clbits: {self.clbits}."
+                )
```

Comment:
When this merges, I'll update #7255 to replace all these checks (and the same in `if_else` and `while_loop`) with a single one to the new condition-resolution mechanism, then it'll be guaranteed to be consistent with `c_if` and remove some duplication.  That's out of scope for this PR, though.